ports/geckolib/wrapper.rs

```diff
@@ -334,20 +348,15 @@ impl<'le> TElement for GeckoElement<'le> {
     }
 
     fn style_attribute(&self) -> &Option<PropertyDeclarationBlock> {
-        panic!("Requires signature modification - only implemented in stylo branch");
-        /*
-        // FIXME(bholley): We should do what Servo does here. Gecko needs to
-        // call into the Servo CSS parser and then cache the resulting block
-        // in the nsAttrValue. That will allow us to borrow it from here.
-        let attr = self.get_attr(&ns!(), &atom!("style"));
-        // FIXME(bholley): Real base URL and error reporter.
-        let base_url = &*DUMMY_BASE_URL;
-        // FIXME(heycam): Needs real ParserContextExtraData so that URLs parse
-        // properly.
-        let extra_data = ParserContextExtraData::default();
-        attr.map(|v| parse_style_attribute(&v, &base_url, Box::new(StdoutErrorReporter),
-                                           extra_data))
-        */
+        unsafe {
+            let ptr = Gecko_GetServoDeclarationBlock(self.element);
+            if ptr.is_null() {
+                &NO_STYLE_ATTRIBUTE
+            } else {
+                let declarations: &GeckoDeclarationBlock = transmute(ptr);
```

Comment:
And .as_ref() will map nicely to the NO_STYLE_ATTRIBUTE case.

In general transmute is a big hammer, and we should restrict our uses to the cases where we really need it.